qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java

```diff
@@ -112,6 +113,43 @@ public void understandsMethodReferences() throws Exception {
     }
 
     /**
+     * PMDValidator does not think that constant is unused when it is used
+     * just from the inner class.
+     * @throws Exception If something wrong happens inside.
+     */
+    @Test
+    public void doesNotComplainAboutConstantsInInnerClasses() throws Exception {
+        // @checkstyle MultipleStringLiteralsCheck (10 lines)
+        final Environment env = new Environment.Mock().withFile(
+            "src/main/java/foo/Foo.java",
+            Joiner.on('\n').join(
+                "package foo;",
+                "interface Foo {",
+                "  final class Bar implements Foo {",
+                "    private static final Pattern TEST =",
+                "      Pattern.compile(\"hey\");",
+                "    public String doSomething() {",
+                "      return Foo.Bar.TEST.toString();",
+                "    }",
+                "  }",
+                "}"
+            )
+        );
+        final StringWriter writer = new StringWriter();
+        Logger.getRootLogger().addAppender(
```

Comment:
@mkordas you should also remove the appender after test, and `writer.flush()` will help in some cases, so it is better to add it.